### ISSUE TITLE
[Feature Request] Batch evaluation for grid interpolation GPs

### ISSUE THREAD

**Issue by Balandat** [cornellius-gp/gpytorch#779]
# 🚀 Feature Request

Evaluating a GP with grid interpolation strategy (e.g. as in the [KISSGP](https://github.com/cornellius-gp/gpytorch/blob/master/examples/05_Scalable_GP_Regression_Multidimensional/KISSGP_Deep_Kernel_Regression_With_LOVE_Fast_Variances_CUDA.ipynb) and [SVGP](https://github.com/cornellius-gp/gpytorch/blob/master/examples/05_Scalable_GP_Regression_Multidimensional/SVDKL_Regression_GridInterp_CUDA.ipynb)) tutorials currently fails in batch mode (e.g. if dimensions of the test data are `b x q x d` rather than `q x d`). 
I'm not super familiar with the interpolation approaches, how hard would it be to support batch evaluation here? Looks like the [GridInterpolationKernel](https://github.com/cornellius-gp/gpytorch/blob/master/gpytorch/kernels/grid_interpolation_kernel.py#L117-L128) has some support for batched inputs. 

## Motivation

I'd like to batch evaluate scalable GPs for BayesOpt 

## Pitch

Ideally the batch evaluation semantics are the same as for ExactGP.

**Comment by jacobrgardner** [comment]
The SVGP example failing in batch mode is strange, since it's the underpinning for e.g. the deep GP implementation on the `deep_gp_wip` branch, which by definition utilizes batch GPs. Is it trivial to get it to fail, or does it fail in specific use cases? Either way, I'll take a look at this -- it shouldn't be failing I think.

**Comment by Balandat** [comment]
Yeah it's trivial - well sort of. The `BatchNorm1d` module has a weird way of handling batched inputs so i replaced it with 
```
class ProperBatchNorm1d(torch.nn.BatchNorm1d):
    
    def forward(self, x):
        if x.dim() > 2:
            return super().forward(x.permute(-2, -1, 0)).permute(-1, 0, -2)
        return super().forward(x)`
```

Then you can just run through the SVGP notebook. Running `model(test_x.unsqueeze(0))` will make it fail with a `RuntimeError` caused by a size mismatch in `gpytorch/utils/interpolation.py`.

Similar thing for the KISSGP tutorial, if you run `model(test_x[:10].unsqueeze(0))`instead of `model(test_x[:10])` you'll get a `RuntimeError: found negative index -9223372036854775808 for dim 1` in `gpytorch/utils/interpolation.py`

**Comment by gpleiss** [comment]
This should be fixed as of #903
